src/lib/krb5/ccache/cc_retr.c

```diff
@@ -162,41 +159,52 @@ pref (krb5_enctype my_ktype, int nktypes, krb5_enctype *ktypes)
  */
 
 krb5_boolean
-krb5int_cc_creds_match_request(krb5_context context, krb5_flags whichfields, krb5_creds *mcreds, krb5_creds *creds)
+krb5int_cc_creds_match_request(krb5_context context, krb5_flags whichfields,
+                               krb5_creds *mcreds, krb5_creds *creds)
 {
-    /* Only match a user-to-user credential if explicitly asked for. */
-    if (! set(KRB5_TC_MATCH_IS_SKEY) && creds->is_skey)
+    krb5_boolean is_skey;
+
+    if ((whichfields & KRB5_TC_MATCH_SRV_NAMEONLY)) {
+        if (!srvname_match(context, mcreds, creds))
+            return FALSE;
+    } else {
+        if (!standard_fields_match(context, mcreds, creds))
```

Comment:
`else if`?

Reply:
The idea is to keep the two if statements parallel, but I agree that it causes a double-take.

Thinking about it, maybe there should be a single helper function (match_princs) that accepts whichfields as a parameter. I'll try that out.